js/page-specific/baltcogo-app/BaltCoGoReporterCtrl.js

```diff
@@ -240,36 +241,42 @@
 			});
 		};
 
-		self.nextClick = function nextClick() {
-			if (validatePanel()) {
-				self.page += 1;
+		function isLocationPage() {
+			return self.page === LOCATION_PAGE_NUMBER;
+		}
 
-				if (self.page === 2 && self.category.name === 'Website Issue') {
-					self.latitude = CONSTANTS.locations.courtHouse.latitude;
-					self.longitude = CONSTANTS.locations.courtHouse.longitude;
-					self.page = 3;
-				}
+		function navigateFormSteps(steps) {
+			self.page += steps;
+			var isLocation = isLocationPage();
+			var isLocationRequired = self.shouldRequireLocation();
+			var shouldSkipLocation = isLocation && !isLocationRequired;
 
-				if (self.page === 2) {
-					$timeout(mapResize, 500);
-				}
-			} else { $scope.citySourcedReporterForm.$setSubmitted(); }
-		};
+			if (shouldSkipLocation) {
+				var pageToShow = self.page + steps;
+				skipLocationPage(pageToShow);
+			}
+		}
 
-		self.prevClick = function prevClick() {
-			self.page -= 1;
+		function skipLocationPage(pageToShow) {
+			self.latitude = CONSTANTS.locations.courtHouse.latitude;
+			self.longitude = CONSTANTS.locations.courtHouse.longitude;
+			self.page = pageToShow;
 
-			if (self.page === 2 && self.category.name === 'Website') {
-				self.latitude = CONSTANTS.locations.courtHouse.latitude;
-				self.longitude = CONSTANTS.locations.courtHouse.longitude;
-				self.page = 1;
-			}
+			$timeout(mapResize, 500);
```

Comment:
`$timeout(mapResize, 500);` needs to happen when the location page loads. Right now it looks like that only happens if you're skipping the location page.

Reply:
@mxsnyder good catch, done.